### PR TITLE
Allow owner to set a discount per buyer

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

File: contracts/MANACrowdsale.sol [hunk 1]
```diff
@@ -42,17 +45,29 @@ contract MANACrowdsale is ContinuousCrowdsale, CappedCrowdsale, WhitelistedCrowd
...
-        // whitelisted investors can purchase at preferential price before crowdsale ends
-        if (isWhitelisted(msg.sender) && !hasEnded()) {
-            return preferentialRate;
-        }
+    function setBuyerRate(address beneficiary, uint256 rate) onlyOwner {
+        require(beneficiary != 0);
+        require(rate != 0);
+
+        buyerRate[beneficiary] = rate;
+    }
```
Comment (maraoz) [inline contracts/MANACrowdsale.sol]: rate should probably be set for `investor`s not `beneficiary`s, and I'd do it in the `WhitelistedCrowdsale#whitelist` function.

Comment (federicobond) [inline contracts/MANACrowdsale.sol]: That clears things up. I was worried that the `beneficiary` / `investor` split would force us to redesign `validPurchase` to accept a beneficiary.

**Review by federicobond — COMMENTED** [review]
I would replace `WhitelistedCrowdsale` with a `PreferentialPriceCrowdsale` variant that stores a map from investor addresses to preferential prices. Any mapping entry with a value greater than 0 is interpreted as a whitelisted buyer. Does that make sense?

**Comment by maraoz** [comment]
:+1:

**Pull request merged**